tests/FunctionsTest.php

```diff
@@ -63,10 +81,147 @@ public function strlenProvider()
     }
 
     /**
-     * @dataProvider strlenProvider
+     * @test
+     * @dataProvider mentionedProvider
      */
-    public function testPolyStrlen($expected, $string)
+    public function testMentioned($part, $attributes, $outcome): void
     {
-        $this->assertEquals($expected, poly_strlen($string));
+        $mockDiscord = getMockDiscord();
```

Comment:
I'm not sure why the same code is not used elsewhere

Reply:
Not sure what you mean by this

Reply:
such as
```php
return [
	[new User(getMockDiscord(), ['id' => '12345']), '12345'],
	[new Channel(getMockDiscord(), ['id' => '12345']), '12345'],
	[new Role(getMockDiscord(), ['id' => '12345']), '12345'],
	['12345', '12345'],
	[null, null],
];
```

```
$mockDiscord = getMockDiscord();

return [
	[new User($mockDiscord, ['id' => '12345']), '12345'],
	[new Channel($mockDiscord, ['id' => '12345']), '12345'],
	[new Role($mockDiscord, ['id' => '12345']), '12345'],
	['12345', '12345'],
	[null, null],
];
```

Reply:
Tbh I never thought about adding more logic then just a single return in a `dataProvider`